bounded_counter/count_min_sketch.py

```diff
@@ -17,71 +17,148 @@ def hash_key(row, key):
 class CountMinSketch(object):
     """
     Data structure used to estimate frequencies of elements in massive data sets with fixed memory footprint.
+
+Example:
+        {
+            cms = CountMinSketch(512) # Use 512 MB
+            print(cms.width)  # 16 777 216
+            print(cms.depth)   # 8
+            print(cms.size)  # 536 870 912 (512 MB in bytes)
+
+            cms.increment("foo")
+            cms.increment("bar")
+            cms.increment("foo")
+
+            print(cms["foo"]) # 2
+            print(cms["bar"]) # 1
+            print(cms.cardinality()) # 2
+            print(cms.sum) # 3
+        }
+
     To calculate memory footprint:
         ( width * depth * cell_size ) + HLL size
     Cell size is
        4B for basic / conservative algorithm
        1B for logcounter8
        2B for logcounter1024 / logcons1024
        HLL size is 32 KB
-    Example:
+    Memory example:
         width 2^26 (67 108 864), depth 4, logcons1024 (2B) has 2^(26 + 2 + 1) + 32 768 = 536.9 MB
         Can be pickled to disk with this exact size
-
-    When choosing parameters, decide on a small depth such as 4 and choose as large width as possible. Increases to width
-    are far more effective than increases to depth.
     """
 
-    def __init__(self, width=16, depth=8, algorithm=''):
+    def __init__(self, size_mb=64, width=None, depth=None, algorithm='conservative'):
         """
         Initializes the Count-Min Sketch structure with the given parameters
-        :param width (int) controls the number of hash buckets in one row of the table, should always be power of 2.
+        :param size_mb (int) controls the maximum size of the Count-min Sketch table.
```

Comment:
We use Google-style docstrings (I sent a link earlier).

Reply:
But google-style only for gensim, re-structured text is nice I think.